port/linux/omrvmem.c

```diff
@@ -483,15 +483,15 @@ findAvailableMemoryBlockNoMalloc(struct OMRPortLibrary *portLibrary, ADDRESS sta
 							Trc_PRT_vmem_omrvmem_findAvailableMemoryBlockNoMalloc_parseDashFailed(lineBuf);
 							break;
 						}
-						end = (uintptr_t)strtoull(next, &next, 16);
+						end = strtoull(next, &next, 16);
 						if ((ULLONG_MAX == end) && (ERANGE == errno)) {
 							dataCorrupt = TRUE;
 							Trc_PRT_vmem_omrvmem_findAvailableMemoryBlockNoMalloc_parseSecondAddressFailed(lineBuf);
 							break;
 						}
 
-						currentMmapRange.start = (ADDRESS)start;
-						currentMmapRange.end = (ADDRESS)end;
+						currentMmapRange.start = (ADDRESS)(uintptr_t)start;
+						currentMmapRange.end = (ADDRESS)(uintptr_t)end;
```

Comment:
Can we avoid the `(uintptr_t)` cast? Is a direct cast to `ADDRESS == (void *)` sufficient?
```suggestion
						currentMmapRange.start = (ADDRESS)start;
						currentMmapRange.end = (ADDRESS)end;
```

Reply:
It's likely the compiler will emit a warning if `sizeof(unsigned long long)` is different than `sizeof(void*)`. Casting to `uintptr_t` as an intermediate step forces truncation to happen correctly and should silence any warnings.